src/Login.tsx

```diff
@@ -29,11 +42,33 @@ export default function Login() {
     return Object.keys(newErrors).length === 0;
   };
 
-  const handleSubmit = (event: React.FormEvent) => {
+  const handleSubmit = async (event: React.FormEvent) => {
     event.preventDefault();
     if (validateFields()) {
-      console.log('Email:', email);
-      console.log('Password:', password);
+      try {
+        const { data } = await login({ variables: { email, password } });
+        const token = data?.login?.token;
+
+        if (token) {
+          localStorage.setItem('authToken', token);
+          setSuccessMessage('Login bem-sucedido!');
+          setLoginError(null);
+        }
+      } catch (err) {
+        if (err instanceof ApolloError) {
+          const graphqlErrors = err.graphQLErrors.map((error) => ({
+            name: error.extensions?.code || 'Error',
+            message: error.message || 'Erro inesperado',
```

Comment:
```suggestion
            name: error.extensions?.code ?? 'Error',
            message: error.message ?? 'Erro inesperado',
```

O operador `??` (nullish coalescing) é ligeiramente diferente ao "or" (`||`) mas é mas próprio pra essa utilização que vc quer: caso o valor da esquerda resulte em null/undefined, ele usa o da direita.